### PR TITLE
Debug/add a log handler

### PR DESCRIPTION
**Pull request by scassard**
This PR aims at creating a Log handler for uplift. It dumps `UnityEngine.Debug.Log` console logs messages to an `uplift.log` file in order to make Uplift debugging easier.

### CODE REVIEW

File: Assets/Plugins/Editor/Uplift/UpliftManager.cs [hunk 1]
```diff
@@ -91,8 +94,14 @@ public struct DependencyState
 
 		public void InstallDependencies(InstallStrategy strategy = InstallStrategy.UPDATE_LOCKFILE)
 		{
-			PackageRepo[] targets = GetTargets(GetDependencySolver(), strategy);
-			InstallPackages(targets);
+			using (LogHandler LH = new LogHandler(appendToCurrentLogFile: true, showStackTrace: false))
+			{
+				Debug.Log("> Install Dependenies with strategy " + strategy);
```
Comment (lacostej) [inline Assets/Plugins/Editor/Uplift/UpliftManager.cs]: you might want to move those 3 log calls into the 2 methods

File: Assets/Plugins/Editor/Uplift/UpliftManager.cs [hunk 2]
```diff
@@ -706,12 +748,14 @@ public void UpdatePackage(PackageRepo newer, bool updateDependencies = true)
 			{
 				using (TemporaryDirectory td = newer.Repository.DownloadPackage(newer.Package))
 				{
+					Debug.Log("Requiered version is not installed, updating package");
```
Comment (lacostej) [inline Assets/Plugins/Editor/Uplift/UpliftManager.cs]: tyop :)

File: Assets/Plugins/Editor/Uplift/LogAggregator.cs [hunk 3]
```diff
@@ -53,6 +53,7 @@ class LogHandlerUtils {
 
 	class LogAggregator : ILogHandler, IDisposable
 	{
+		//TODO reactivate Log aggregator
```
Comment (lacostej) [inline Assets/Plugins/Editor/Uplift/LogAggregator.cs]: where was it deactivated?

File: Assets/Plugins/Editor/Uplift/LogHandler.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+using UnityEngine;
+using System.Collections;
+using System.IO;
+using System.Text;
+using System;
+
+public class LogHandler : IDisposable
+{
+	public bool ShowStackTrace = false;
```
Comment (niezbop) [inline Assets/Plugins/Editor/Uplift/LogHandler.cs]: naming: `showStackTrace`! camelCasing. Same for the following vars

File: Assets/Plugins/Editor/Uplift/LogHandler.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		Application.logMessageReceived -= HandleLog;
+
+		if (OutputStream != null)
+		{
+			OutputStream.Close();
+			OutputStream = null;
+		}
+	}
+
+	void HandleLog(string logString, string stackTrace, LogType type)
```
Comment (niezbop) [inline Assets/Plugins/Editor/Uplift/LogHandler.cs]: We lose the `type` information. Not sure it's super important but maybe we want to prefix the logfile with something like `[type]`, `type.firstCharacter` or something like that

File: Assets/Plugins/Editor/Uplift/UpliftManager.cs (hunk 1 above)
Comment (niezbop) [inline Assets/Plugins/Editor/Uplift/UpliftManager.cs]: typo

File: Assets/Plugins/Editor/Uplift/LogAggregator.cs (hunk 3 above)
Comment (scassard) [reply Assets/Plugins/Editor/Uplift/LogAggregator.cs]: I deactivated it the the first commit ["Lots of debug log"](https://github.com/DragonBox/uplift/pull/84/commits/48dee360a847cafd9ad7f5d143051a0cb7195d61), and was reactivated in ["Reactivate log aggregator which was disabled for debugging purposes"](https://github.com/DragonBox/uplift/pull/84/commits/c55c9f677778098ae792c5e777262c3c8a204eb4l).

I forgot to mention I deactivated it in the first commit, my bad.

File: Assets/Plugins/Editor/Uplift/UpliftManager.cs [hunk 6]
```diff
@@ -91,10 +94,14 @@ public struct DependencyState
...
 		{
-			PackageRepo[] targets = GetTargets(GetDependencySolver(), strategy);
+			using (LogHandler LH = new LogHandler(appendToCurrentLogFile: true, showStack: false))
+			{
+				Debug.Log("Install Dependencies with strategy " + strategy);
+			  PackageRepo[] targets = GetTargets(GetDependencySolver(), strategy);
 
-			bool updateLockfile = (strategy != InstallStrategy.ONLY_LOCKFILE);
-			InstallPackages(targets, updateLockfile);
+			  bool updateLockfile = (strategy != InstallStrategy.ONLY_LOCKFILE);
```
Comment (lacostej) [inline Assets/Plugins/Editor/Uplift/UpliftManager.cs]: weird indentation in this file

**Pull request merged**